File: Core/Entities/Publisher.cs
using System;
using System.Collections.Generic;

namespace Core.Entities
{
    public class Publisher : BaseEntity
    {
        public string FirstName { get; set; }
        public string MiddleName { get; set; }
        public string LastName { get; set; }
        public DateTime BaptismDate { get; set; }
        public DateTime BirthDate { get; set; }
        public string PhotoUrl { get; set; }
        public string Gender { get; set; }
        public int PhoneNumber { get; set; }
        public IEnumerable<AppointedPublisher> AppointedPublishers { get; set; }
        public int StatusId { get; set; }
        public Group Group { get; set; }
        public int GroupId { get; set; }
        public IEnumerable<Report> Reports { get; set; }
    }
}

File: client/src/app/shared/models/publisher.ts
import { IDictionary } from './dictionary';

export interface IPublisher {
  id: number;
  firstName: string;
  middleName: string;
  lastName: string;
  age: number;
  baptismDate: Date;
  birthDate: Date;
  photoUrl: string;
  gender: string;
  status: IDictionary;
  group: string;
}

export interface IPostPublisher {
  id:number;
  firstName?: any;
  middleName?: any;
  lastName?: any;
  baptismDate: string;
  birthDate: string;
  photoUrl?: any;
  gender?: any;
  groupId: number;
}
